### PR TITLE
Added useRole

### LINKED ISSUE #7: Add: useRole hook
## Reference

https://floating-ui.com/docs/useRole

## Description

Adds base screen reader props to the reference and floating elements for a given role. This is useful to automatically apply ARIA props to the reference and floating elements to ensure they’re accessible to assistive technology, including item elements if narrowly specified.

## Props

- [x] `enabled`
- [x] `role`
- [x] `Component roles`
- [x] `Native roles`

### PR DESCRIPTION
**Pull request by Hugos68**
Closes #7 

TODO:
- [x] Write docs
- [x] Write tests (delayed until #43 is merged)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,7 +123,29 @@ This will ensure all event handlers will be registered rather being overruled by
...
 ### useRole
 
-(tbd)
+#### Usage
+
+```html
+<script>
+	import { useFloating, useInteractions, useRole } from '@skeletonlabs/floating-ui-svelte';
+
+	const floating = useFloating();
```
Comment (endigo9740) [inline README.md:132]: Remove the extra line breaks between `const` declarations to save vertical space.

File: README.md [hunk 2]
```diff
@@ -123,7 +123,29 @@ This will ensure all event handlers will be registered rather being overruled by
...
+<button {...interactions.getReferenceProps()}>Reference</button>
+<div {...interactions.getFloatingProps()}>Tooltip</div>
+```
+
+#### Options
+
+| Property | Description | Type | Default Value |
+| -------- | ----------- | ---- | ------------- |
+| enabled | Enables the interaction | boolean | true |
+| role | The role that the floating element should be | AriaRole | 'dialog' |
```
Comment (endigo9740) [inline README.md:149]: We may want to link to the Floating UI docs for these two subsections. Seems like useful info that is not strictly speaking React-specific.

- https://floating-ui.com/docs/useRole#component-roles
- https://floating-ui.com/docs/useRole#native-roles

File: README.md [hunk 3]
```diff
@@ -123,7 +123,29 @@ This will ensure all event handlers will be registered rather being overruled by
...
+	const role = useRole(floating.context, { role: 'tooltip' });
+
+	const interactions = useInteractions([role]);
+</script>
+
+<button {...interactions.getReferenceProps()}>Reference</button>
+<div {...interactions.getFloatingProps()}>Tooltip</div>
+```
+
+#### Options
```
Comment (endigo9740) [inline README.md:143]: Floating UI calls these "Props", not "Options". Though I think options or parameters fit better. Use your best judgement here.

https://floating-ui.com/docs/useRole#props

**Review by endigo9740 — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (Hugos68) [reply README.md:143]: I'll leave this for now since the entire project uses this terminology. We can handle this in a seperate PR if this is an issue.

**Review by endigo9740 — APPROVED** [review]

**Pull request merged**